### PR TITLE
OHRM5X-1677: Develop Widget - My Actions Summary View [BE]

### PR DESCRIPTION
**Pull request by SandeepaOHRM**

### CODE REVIEW

**Review by RajithaKumara — CHANGES_REQUESTED** [review]

**Review by Super-Chama — APPROVED** [review]

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeActionSummaryAPI.php [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+        /**
+         * Pending Leave Requests
+         */
+
+        $leaveRequestSearchFilterParams = new LeaveRequestSearchFilterParams();
+        $leaveRequestSearchFilterParams->setEmpNumbers(array_values($accessibleEmpNumbers));
+        $leaveRequestSearchFilterParams->setIncludeEmployees(
+            LeaveRequestSearchFilterParams::INCLUDE_EMPLOYEES_ONLY_CURRENT
+        );
+        $leaveRequestSearchFilterParams->setStatuses([Leave::LEAVE_STATUS_LEAVE_PENDING_APPROVAL]);
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmDashboardPlugin/Api/EmployeeActionSummaryAPI.php]: moved

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeActionSummaryAPI.php [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+     */
+    public function getValidationRuleForGetOne(): ParamRuleCollection
+    {
+        $paramRules = new ParamRuleCollection(
+            $this->getValidationDecorator()->notRequiredParamRule(
+                new ParamRule(
+                    CommonParams::PARAMETER_EMP_NUMBER,
+                    new Rule(Rules::IN_ACCESSIBLE_EMP_NUMBERS)
+                ),
+            ),
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmDashboardPlugin/Api/EmployeeActionSummaryAPI.php]: done

File: src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php [hunk 3]
```diff
@@ -208,42 +208,61 @@ private function getPerformanceReviewQueryBuilderWrapper(PerformanceReviewSearch
         $qb->andWhere($qb->expr()->eq('reviewGroup.name', ':reviewGroupName'))
             ->setParameter('reviewGroupName', ReviewerGroup::REVIEWER_GROUP_SUPERVISOR);
 
-        if (!is_null($performanceReviewSearchFilterParams->getReviewerEmpNumber())) {
-            $qb->andWhere($qb->expr()->eq('reviewerEmployee.empNumber', ':supervisorEmpNumber'))
-                ->setParameter('supervisorEmpNumber', $performanceReviewSearchFilterParams->getReviewerEmpNumber());
-        }
+        if ($performanceReviewSearchFilterParams->isForPendingReviewActionWidget()) {
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmPerformancePlugin/Dao/PerformanceReviewDao.php]: yes, refactored

File: src/plugins/orangehrmDashboardPlugin/Api/EmployeeActionSummaryAPI.php [hunk 4]
```diff
@@ -0,0 +1,183 @@
...
+    /**
+     * @inheritDoc
+     */
+    public function getOne(): EndpointResult
+    {
+        $empNumber = $this->getAuthUser()->getEmpNumber();
+
+        $accessibleEmpNumbers = $this->getUserRoleManager()->getAccessibleEntityIds(Employee::class);
+        $authUserIndex = array_search($empNumber, $accessibleEmpNumbers);
+        unset($accessibleEmpNumbers[$authUserIndex]);
```
[the remark this answers is not in the document]
Comment (SandeepaOHRM) [reply src/plugins/orangehrmDashboardPlugin/Api/EmployeeActionSummaryAPI.php:54]: sure

**Pull request merged**